### PR TITLE
Fix issue when right-click + left-click starts a widget drag interaction

### PR DESCRIPTION
**Pull request by KaroAug**
## Changes

Fixed the issue when right-click + left-click starts a widget drag interaction.

## Testing

Tested via test-app.

### CODE REVIEW

File: ui/appui-react/src/appui-react/layout/base/DragManager.tsx [hunk 1]
```diff
@@ -531,7 +531,7 @@ export function DragProvider(props: DragProviderProps) {
   const dragManager = React.useRef(new DragManager());
   React.useEffect(() => {
     const mouseMove = (e: MouseEvent) => {
-      dragManager.current.handleDrag(e.clientX, e.clientY);
+      if (e.buttons !== 2) dragManager.current.handleDrag(e.clientX, e.clientY);
```
Comment (GerardasB) [inline ui/appui-react/src/appui-react/layout/base/DragManager.tsx:534]: Could we move this check to mouse/pointer down instead?

File: ui/appui-react/src/appui-react/layout/widget/Widget.tsx [hunk 2]
```diff
@@ -159,12 +159,21 @@ export const Widget = React.forwardRef<HTMLDivElement, WidgetProps>(
...
+      },
+      []
+    );
     const ref = useRefs(forwardedRef, elementRef);
     const className = classnames("nz-widget-widget", props.className);
     return (
       <WidgetContext.Provider value={widgetContextValue}>
         <div
           className={className}
+          onContextMenu={handleContextMenu}
```
Comment (GerardasB) [inline ui/appui-react/src/appui-react/layout/widget/Widget.tsx:176]: Why is this needed, can't we keep the context menu?

**Review by GerardasB — DISMISSED** [review]

File: ui/appui-react/src/appui-react/layout/base/DragManager.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (GerardasB) [reply ui/appui-react/src/appui-react/layout/base/DragManager.tsx]: Usually not an issue, I've tested other projects i.e. for resize behavior.

[the remark this answers is not in the document]
Comment (GerardasB) [reply ui/appui-react/src/appui-react/layout/base/DragManager.tsx]: Whether to start the drag interaction or not is handled in mouse down only.

File: ui/appui-react/src/appui-react/layout/widget/Widget.tsx (hunk 2 above)
Comment (GerardasB) [reply ui/appui-react/src/appui-react/layout/widget/Widget.tsx]: Removed this, users can still open a context menu.

**Review by saskliutas — APPROVED** [review]

**Review by ignas-k — APPROVED** [review]

**Comment by GerardasB** [comment]
Fixes #132
